x/staking/types/validator.go

```diff
@@ -439,17 +438,8 @@ func (v Validator) GetOperator() sdk.ValAddress {
 	return addr
 }
 func (v Validator) GetConsPubKey() crypto.PubKey {
-	// The way things are refactored now, v.ConsensusPubkey is sometimes a TM
-	// ed25519 pubkey, sometimes our own ed25519 pubkey. This is very ugly and
-	// inconsistent.
-	// Luckily, here we coerce it into a TM ed25519 pubkey always, as this
-	// pubkey will be passed into TM (eg calling encoding.PubKeyToProto).
 	pk := sdk.MustGetPubKeyFromBech32(sdk.Bech32PubKeyTypeConsPub, v.ConsensusPubkey)
 
-	if intoTmPk, ok := pk.(cryptotypes.IntoTmPubKey); ok {
-		return intoTmPk.AsTmPubKey()
-	}
-
 	return pk
```

Comment:
let's see if all tests will pass after the https://github.com/cosmos/cosmos-sdk/pull/7597 merge

Reply:
unless we merge this one first muhahah lol

Reply:
#7597 got merged ;)

Reply:
dam, haha. looks like I have to undo some changes in that PR, it brought tendermint keys deeper into the sdk